com.kms.cura.dal/src/com/kms/cura/dal/database/DoctorUserDatabaseHelper.java

```diff
@@ -0,0 +1,189 @@
+			con.setAutoCommit(true);
+			stmt.close();
+		}
+		return getEntityFromResultSet(rs);
+	}
+
+	private Map<String, Object> getColumnValueMapForEntity(DoctorUserEntity entity) {
+		if (entity == null) {
+			// handle error;
+			return null;
+		}
+		Map<String, Object> columnValueMap = new LinkedHashMap<String, Object>();
+		for (DoctorColumn doctortColumn : DoctorColumn.values()) {
+			String columnName = doctortColumn.getColumnName();
+			switch (doctortColumn) {
+			case BIRTH:
+				columnValueMap.put(columnName, entity.getBirth());
+				break;
+			case GENDER:
+				columnValueMap.put(columnName, entity.getGender());
+				break;
+			case INSURANCE:
+				columnValueMap.put(columnName, entity.getInsurance());
+				break;
+			case LOCATION:
+				columnValueMap.put(columnName, entity.getLocation());
+				break;
+			case NAME:
+				columnValueMap.put(columnName, entity.getName());
+				break;
+			case USER_ID:
+				columnValueMap.put(columnName, entity.getId());
+				break;
+			case DEGREE_ID:
+				columnValueMap.put(columnName, entity.getDegree().getId());
+				break;
+			case EXPERIENCE:
+				columnValueMap.put(columnName, entity.getExperience());
+				break;
+			case MAX_PRICE:
+				columnValueMap.put(columnName, entity.getMaxPrice());
+				break;
+			case MIN_PRICE:
+				columnValueMap.put(columnName, entity.getMinPrice());
+				break;
+			case PHONE:
+				columnValueMap.put(columnName, entity.getPhone());
+				break;
+			case RATING:
+				columnValueMap.put(columnName, entity.getRating());
+				break;
+			}
+		}
+		return columnValueMap;
+	}
+
+	public PreparedStatement getInsertStatementForUser(DoctorUserEntity entity) throws SQLException {
+		Map<String, Object> doctorValueMap = getColumnValueMapForEntity(entity);
+		String databaseName = "Doctor";
+		String databaseMethod = "INSERT INTO";
+		return createPreparedStatement(doctorValueMap, databaseName, databaseMethod);
+	}
+
+	protected void insertToDoctorReferenceTables(DoctorUserEntity entity) throws SQLException {
+		List<ReferenceTableRow> referenceRows = new ArrayList<ReferenceTableRow>();
+		for (FacilityEntity facility : entity.getFacility()) {
+			ReferenceTableRow referenceRow = new ReferenceTableRow();
+			referenceRow.setFirstRefKey("facility_id");
+			referenceRow.setFirstValue(facility.getId());
+			referenceRow.setSecondRefKey("doctor_id");
+			referenceRow.setSecondValue(entity.getId());
+			referenceRows.add(referenceRow);
+		}
+		insertReferenceRowsToReferenceTable("Doctor_Facilities", referenceRows);
+
+		referenceRows.clear();
+		for (SpecialityEntity speciality : entity.getSpeciality()) {
+			ReferenceTableRow referenceRow = new ReferenceTableRow();
+			referenceRow.setFirstRefKey("speciality_id");
+			referenceRow.setFirstValue(speciality.getId());
+			referenceRow.setSecondRefKey("doctor_id");
+			referenceRow.setSecondValue(entity.getId());
+			referenceRows.add(referenceRow);
+		}
+		insertReferenceRowsToReferenceTable("Doctor_Specialties", referenceRows);
```

Comment:
Same here, should create DoctorSpecialityColumn to hold "speciality_id" and "doctor_id"